controllers/operandrequest/operandrequest_controller_test.go

```diff
@@ -162,17 +182,31 @@ var _ = Describe("OperandRegistry controller", func() {
 
 			By("Disabling the etcd operator")
 			Eventually(func() bool {
-				requestInstance := &operatorv1alpha1.OperandRequest{}
-				Expect(k8sClient.Get(ctx, requestKey, requestInstance)).Should(Succeed())
-				requestInstance.Spec.Requests[0].Operands = requestInstance.Spec.Requests[0].Operands[1:]
-				Expect(k8sClient.Update(ctx, requestInstance)).Should(Succeed())
+				requestInstance1 := &operatorv1alpha1.OperandRequest{}
+				Expect(k8sClient.Get(ctx, requestKey1, requestInstance1)).Should(Succeed())
+				requestInstance1.Spec.Requests[0].Operands = requestInstance1.Spec.Requests[0].Operands[1:]
+				Expect(k8sClient.Update(ctx, requestInstance1)).Should(Succeed())
+				requestInstance2 := &operatorv1alpha1.OperandRequest{}
+				Expect(k8sClient.Get(ctx, requestKey2, requestInstance2)).Should(Succeed())
+				requestInstance2.Spec.Requests[0].Operands = requestInstance2.Spec.Requests[0].Operands[1:]
+				Expect(k8sClient.Update(ctx, requestInstance2)).Should(Succeed())
 				etcdCluster := &v1beta2.EtcdCluster{}
 				err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: "example", Namespace: operatorNamespaceName}, etcdCluster)
 				return err != nil && errors.IsNotFound(err)
 			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
 
-			By("Deleting the OperandRequest")
-			Expect(k8sClient.Delete(ctx, request)).Should(Succeed())
+			By("Deleting the first OperandRequest")
+			Expect(k8sClient.Delete(ctx, request1)).Should(Succeed())
+
+			By("Checking jenkins operator has not been deleted")
+			Eventually(func() error {
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
+				return err
+			}, testutil.Timeout, testutil.Interval).Should(BeNil())
```

Comment:
I created two `OperandRequest` in unit test and both of them have operands `etcd` and `jenkins`, but from different registry.
So if I delete one of the `OperandRequest`. the `etcd` and `jenkins` operator should remain in the cluster. So the `err` returned should `nil` I believe.

It is confusing to me that when I try `return err==nil`, Should(BeFalse())` still passes the test.

And at line 216, after the last `OperandRequest` is deleted, `err` should `isNotFound`.

Reply:
Since it will retry several times, is it possible in the first the CSV exist but after waiting for a few, it is delered